### ISSUE TITLE
Connect4 Model not learning

### ISSUE THREAD

**Issue by xuehy** [suragnair/alpha-zero-general#96]
Hi. I am training the model for Connect4. The  model update is decided by self-play with previous models. If it reaches certain winning rate(which is 0.6 in the code, then the model updates. 

However, in my situations, after the training is started, the winning rate never reaches 0.6. And the model gets stuck at the first iteration. 

Is it usual that the first iteration has such difficulty to update the model? After the first iteration of training, the model always gets worse so that it cannot be updated, so what can be wrong?

**Comment by imrehg** [comment]
How many iterations you have done? I'm training Connect4 (with PyTorch, based on the network from Othello), and it is clearly progressing. Though Connect4 might be an odd one out of the games, as it is solved (the first player always wins if playing perfectly, see [this thesis](http://www.informatik.uni-trier.de/~fernau/DSL0607/Masterthesis-Viergewinnt.pdf)). Thus in a model that implements that the perfect game would come with 50-50% win ratio.

Anyways, I think lowering that rate a bit, and increasing the plays to check on (so finer-grained differences can be seen) might be the way? Just a hunch, as I'm a total noob that is trying things out and learning along the way.

But would be interested in what others think...

**Comment by xuehy** [comment]
I have modified the code so that the model is forced to be updated after every iteration(generation). I also increase the epochs for each iteration to 10. However, the loss does not decrease even after thousands of batches.If I reduce the epochs for each iter to 1 or 2, the training loss decreases at first, but after 100+ iterations, the loss again increases. I feel really confused. Anyone suceeded in training model  for CONNECT4?

**Comment by suragnair** [comment]
#156 may explain this phenomena. If you can, please give it a shot! Thanks.
